### PR TITLE
Update go dependencies

### PR DESCRIPTION
**Pull request by jim**
## Description

It seems that in the switch from `dep` to `go mod` a bunch of our dependencies were downgraded, probably due to how `go mod` uses the [minimum version that will satisfy requirements for indirect dependencies](https://github.com/golang/go/wiki/Modules#version-selection).

This PR updates most of our dependencies to the latest version.

## Reviewer Notes

You can run the script to update dependencies locally like so:

```console
$ make go_deps_update
```

I noticed that `go vet` sometimes alters `go.mod`. I'm not sure why.

## Code Review Verification Steps

* [ ] Request review from a member of a different team.

## References

* [PR where we switched to go mod](https://github.com/transcom/mymove/pull/1932)

### CODE REVIEW

File: docs/how-to/manage-dependencies-with-go-mod.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+## Update all go dependencies
+
+```console
+$ go get -u
+```
+
+## Update a specific dependency
+
+```console
+$ go get -u github.com/pkg/errors
```
Comment (chrisgilmerproj) [inline docs/how-to/manage-dependencies-with-go-mod.md]: Would be nice to also have an example of getting the dependency `@master`, which I think is the right syntax. I think a lot of our deps we want at the master version.

**Comment by pjdufour-truss** [comment]
This PR is reverting cobra, viper, and pflag.  Since we want master branch for those, so this is still blocked.

**Comment by jim** [comment]
@pjdufour-truss Can you take a look at this again when you have a minute? I think we're in good shape.

We might need to tag our fork `pdfcpu` to a version as there isn't a nice way to pin to a non-master branch and if possible I'd like to avoid special casing for a single dependency.

**Comment by pjdufour-truss** [comment]
Could you also look into using https://godoc.org/os/exec#CommandContext and maybe setting a default timeout for each package with a few retries?  One of the go get commands stalled on me.

**Comment by jim** [comment]
@pjdufour-truss I didn't add any retries yet, but I did add a timeout as you suggested. If we find ourselves retrying a lot we can go ahead and put that in.

**Comment by pjdufour-truss** [comment]
Can we remove `gosec` (or comment it out with TODO) since it wasn't working with modules yet?

**Comment by pjdufour-truss** [comment]
It looks like the newest version of pop changed UUID libraries.  Can we confirm that's not an issue?  https://github.com/gobuffalo/pop/pull/348

**Comment by jim** [comment]
@pjdufour-truss 

> It looks like the newest version of pop changed UUID libraries. Can we confirm that's not an issue? gobuffalo/pop#348

It looks like we're already on `github.com/gofrs/uuid v3.2.0+incompatible` on master. This is also a different fork of the same original codebase with only minimal API changes. The app and tests run so I think we're OK here.

> Can we remove gosec (or comment it out with TODO) since it wasn't working with modules yet?

Sure thing. I'll make that change. Thanks for mentioning it.

**Pull request merged**